src/java/org/apache/cassandra/db/virtual/PrimaryIdTable.java

```diff
@@ -0,0 +1,350 @@
+ *     PRIMARY KEY ((keyspace_name, table_name), token_value, key)
+ * );
+ * </pre>
+ *
+ * <p>Note:
+ * <ul>
+ *     <li>The `size_estimate` and `sstables` columns represent aggregate information about SSTable sizes and counts.</li>
+ *     <li>Range queries across multiple tables and updates are not supported as this is a read-only table.</li>
+ * </ul>
+ */
+public class PrimaryIdTable implements VirtualTable
+{
+    private static final Logger logger = LoggerFactory.getLogger(PrimaryIdTable.class);
+    public static final String NAME = "primary_ids";
+
+    private static final String TABLE_READ_ONLY_ERROR = "The specified table is read-only.";
+    private static final String UNSUPPORTED_RANGE_QUERY_ERROR = "Range queries are not supported. Please provide both a keyspace and a table name.";
+    private static final String REVERSED_QUERY_ERROR = "Reversed queries are not supported.";
+    private static final String KEYSPACE_NOT_EXIST_ERROR = "The keyspace '%s' does not exist.";
+    private static final String TABLE_NOT_EXIST_ERROR = "The table '%s' does not exist in the keyspace '%s'.";
+    private static final String KEY_ONLY_EQUALS_ERROR = "The 'key' column can only be used in an equality query for this virtual table.";
+    private static final String KEY_NOT_WITHIN_BOUNDS_ERROR = "The specified 'key' is not within the provided token value bounds.";
+
+    private static final String COLUMN_KEYSPACE_NAME = "keyspace_name";
+    private static final String COLUMN_TABLE_NAME = "table_name";
+    private static final String COLUMN_TOKEN_VALUE = "token_value";
+    private static final String COLUMN_KEY = "key";
+    private static final String COLUMN_SIZE_ESTIMATE = "size_estimate";
+    private static final String COLUMN_SSTABLES = "sstables";
+
+    private final TableMetadata metadata;
+    private final ColumnMetadata sizeEstimateColumn;
+    private final ColumnMetadata sstablesColumn;
+
+    @VisibleForTesting
+    final CopyOnWriteArrayList<Consumer<DecoratedKey>> readListener = new CopyOnWriteArrayList<>();
+
+    public PrimaryIdTable(String keyspace)
+    {
+        this.metadata = TableMetadata.builder(keyspace, NAME)
+                                     .kind(TableMetadata.Kind.VIRTUAL)
+                                     .partitioner(new LocalPartitioner(CompositeType.getInstance(UTF8Type.instance, UTF8Type.instance)))
+                                     .addPartitionKeyColumn(COLUMN_KEYSPACE_NAME, UTF8Type.instance)
+                                     .addPartitionKeyColumn(COLUMN_TABLE_NAME, UTF8Type.instance)
+                                     .addClusteringColumn(COLUMN_TOKEN_VALUE, IntegerType.instance)
+                                     .addClusteringColumn(COLUMN_KEY, UTF8Type.instance)
+                                     .addRegularColumn(COLUMN_SIZE_ESTIMATE, CounterColumnType.instance)
+                                     .addRegularColumn(COLUMN_SSTABLES, CounterColumnType.instance)
```

Comment:
Friday after 5pm... I was not expecting to see counters... ill have to look closer Monday...

Reply:
it doesnt actually use counters in normal sense, its because if theres 2 sstables, the cell resolver for the UnfilteredRowIterator will merge them by adding them together instead of replacing with lww